### PR TITLE
Specify acceptable entities

### PR DESCRIPTION
**Pull request by jayvdb**
# Pull Request Check List

Resolves: https://github.com/Riverside-Healthcare/djLint/issues/156

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: src/djlint/rules.yaml [hunk 1]
```diff
@@ -172,7 +172,7 @@
     message: Do not use entity references.
     flags: re.I
     patterns:
-    - '&.{5};'
+    - '&(?!(lt|gt|amp|quot|nbsp|ensp|emsp|thinsp))[^;]+;'
```
Comment (jayvdb) [inline src/djlint/rules.yaml:175]: I'm guessing you'd want this list of entities to be a config setting. Happy to do that

**Comment by christopherpickering** [comment]
Thanks, looks good, but I think we should remove the `[^;]+` from the regex, or at least exclude whitespace. The rule is catching `&` in text now too.
for example,
```html
<div>
    Nice paragraph & this is also nice;
</div>
```
is triggering the rule.

Maybe something like this: https://regex101.com/r/ADssOm/1

What do you think?

**Comment by jayvdb** [comment]
I think we can use `[a-z]+` ?

Happy to add a test case to prevent catching your example - thx for catching that.

**Comment by christopherpickering** [comment]
Yep, I would still add a max though, `[A-z]{,7}`?

**Comment by jayvdb** [comment]
According to https://dev.w3.org/html5/html-author/charref , there are entities that are 23 chars at least, like `&DiacriticalDoubleAcute;`

**Comment by christopherpickering** [comment]
Oh crazy, I didn't realize! Ok, we'll take it without a limit then :)

**Comment by christopherpickering** [comment]
Looks good, its all set now?

**Comment by jayvdb** [comment]
Yup, I think this is good enough.  If there is an entity with name longer than 30, we'll find out ;-)

**Pull request merged**